jsm.go

```diff
@@ -408,6 +412,11 @@ func (js *js) upsertConsumer(stream, consumerName string, cfg *ConsumerConfig, o
 		}
 		return nil, info.Error
 	}
+
+	// check whether multiple filter subjects (if used) are reflected in the returned ConsumerInfo
+	if len(cfg.FilterSubjects) != 0 && len(info.Config.FilterSubjects) == 0 {
```

Comment:
To test this I guess would need to run against a version of the server that does not support filter subjects 🤔

Reply:
Yup, this is basically server compatibility check.